src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -106,51 +103,31 @@ public ReadOnlyBytes Slice(Range range)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public ReadOnlyBytes Slice(int index, int length)
+        public ReadOnlyBytes Slice(long index, long length)
         {
-            if(length==0) return ReadOnlyBytes.Empty;
+            if(length==0) return Empty;
 
             var first = First;
             if (first.Length >= length + index)
```

Comment:
Will this comparison by between int32 or will first.Length get casted to int64? If it is int32, it could cause integer overflow.

Reply:
It will be Int64 comparison.